packages/requisitions/src/ResponseRequisition/api/hooks.ts

```diff
@@ -105,22 +106,22 @@ export const useResponseFields = <
 interface UseResponseLinesController
   extends SortController<ResponseLineFragment> {
   lines: ResponseLineFragment[];
+  columns: Column<ResponseLineFragment>[];
 }
 
 export const useResponseLines = (): UseResponseLinesController => {
-  const { sortBy, onChangeSortBy } = useSortBy<ResponseLineFragment>({
-    key: 'itemName',
-    isDesc: false,
-  });
   const { lines } = useResponseFields('lines');
+  const { columns, onChangeSortBy, sortBy } = useResponseColumns();
 
   const sorted = useMemo(() => {
-    return (lines?.nodes ?? []).sort(
-      getDataSorter(sortBy.key as keyof ResponseLineFragment, !!sortBy.isDesc)
-    );
-  }, [sortBy, lines]);
-
-  return { lines: sorted, sortBy, onChangeSortBy };
+    const currentColumn = columns.find(({ key }) => key === sortBy.key);
+    const { getSortValue } = currentColumn ?? {};
+    return getSortValue
+      ? lines?.nodes.sort(getColumnSorter(getSortValue, !!sortBy.isDesc))
+      : lines?.nodes;
+  }, [sortBy.key, sortBy.isDesc, lines]);
```

Comment:
With this method of sorting, we re-order whenever the lines change, so when we open the modal and change a quantity, and we have sorted on quantity, the order changes. Is that good or bad?

Reply:
In some ways good - you finish editing and the table is sorted according to the values shown.
In other ways bad - if you OK & Next then you'll get a fairly surprising next line and the nexting will stop before or after you expect it to.
Overall.. ok for now